src/backends/github/implementation.js

```diff
@@ -137,8 +137,21 @@ export default class GitHub {
       const response = await this.api.persistFiles(null, [mediaFile], options);
       const repo = this.repo || this.getRepoFromResponseUrl(response.url);
       const { value, size, path, fileObj } = mediaFile;
-      const url = `https://raw.githubusercontent.com/${repo}/${this.branch}${path}`;
-      return { id: response.sha, name: value, size: fileObj.size, url, path: trimStart(path, '/') };
+      let url = `https://raw.githubusercontent.com/${repo}/${this.branch}${path}`;
+
+      // Assets uploaded to private repos will need valid access tokens.
+      const isPrivateRepo = await this.api.isPrivateRepo();
+      if (isPrivateRepo) {
+        const files = await this.api.listFiles(this.config.get('media_folder'));
+        for (let file of files) {
```

Comment:
I think we could make this code a bit simpler by using `find()` instead of looping:
```js
const file = files.find(f => (f.sha === mediaFile.sha));
if (file) {
  url = file.download_url;
}
```